scripts/properties2json.js

```diff
@@ -26,10 +45,13 @@ function getJSON(locale) {
 }
 
 function processMessageFiles(locales) {
-  return Promise.all(locales.map(getJSON));
+  if (config.supportedLocales !== "*") {
```

Comment:
It looks like this is deciding to use locales or not after getting them.

Flow looks like:

1. `getListLocales` returns locales.
2. check if we want to use those locales or not.

I think we can optimize this by reversing this order. First check if we need all locales or not, then call `getListLocales` by seeing if `config.supportedLocales !== "*"` is true.